src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -929,18 +929,20 @@ struct aie_tiles_stats
 // structure to get aie tiles status raw buffer
 struct aie_tiles_status_info
 {
-  using result_type = boost::any;
+  using result_type = xrt_core::query::aie_tiles_status_info::result_type;
 
   static result_type
   get(const xrt_core::device* device, key_type key, const boost::any& param)
   {
     auto data = boost::any_cast<xrt_core::query::aie_tiles_status_info::parameters>(param);
+    uint32_t cols_filled = 0;
+    uint32_t buf_size = data.col_size * data.num_cols;
 
-    std::vector<char> buf(data.col_size * data.num_cols, 0);
+    std::vector<char> buf(buf_size, 0);
 
-    // TODO : Add code to get the data
+    // TODO : Add code to get the data and cols filled info
 
-    return buf;
+    return {buf, cols_filled};
```

Comment:
Or even more modern
```suggestion
    return { .buf = buf, .cols_filled = cols_filled };
```

Reply:
Hi @keryell, thanks for the suggestion but this feature needs compiler to have C++20 support. XRT at present uses c++17 as default and windows flows also does not support c++20 yet. So let me create structure and return it for now. Thanks.

Reply:
Ah, too bad.